src/commands/createNewProject/ProjectCreateStep/PowerShellProjectCreateStep.ts

```diff
@@ -54,9 +73,97 @@ export class PowerShellProjectCreateStep extends ScriptProjectCreateStep {
             await fse.writeFile(profileps1Path, profileps1);
         }
 
+        const majorVersion: number | undefined = await this.getLatestAzModuleMajorVersion(progress);
+        if (majorVersion !== undefined) {
+            progress.report({
+                message: localize('successfullyConnected', 'Successfully retrieved {0} information from PowerShell Gallery', this.azModuleName)
+            });
+        } else {
+            progress.report({
+                message: localize('failedToConnect', 'Failed to get {0} module version. Edit the requirements.psd1 file when the powershellgallery.com is accessible.', this.azModuleName)
+            });
+        }
+
         const requirementspsd1Path: string = path.join(context.projectPath, requirementspsd1FileName);
         if (await confirmOverwriteFile(requirementspsd1Path)) {
-            await fse.writeFile(requirementspsd1Path, requirementspsd1);
+            if (majorVersion !== undefined) {
+                await fse.writeFile(requirementspsd1Path, requirementspsd1(majorVersion));
+            } else {
+                await fse.writeFile(requirementspsd1Path, requirementspsd1Offine);
+            }
+        }
+    }
+
+    private async getLatestAzModuleMajorVersion(progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<number | undefined> {
+        progress.report({
```

Comment:
Yeah, this is hard to read if a user has good connection. But I think we should still keep the message here though because we want to warn them if they don't have internet connection.

cc: @Francisco-Gamino